projects/observability/src/shared/graphql/request/handlers/entities/query/topology/entity-topology-graphql-query-handler.service.ts

```diff
@@ -154,8 +154,8 @@ export class EntityTopologyGraphQlQueryHandlerService
     nodeMap: Map<string, EntityNode>,
     serverEntity: TopologyEntity
   ): EntityNode {
-    if (nodeMap.has(serverEntity.id)) {
-      return nodeMap.get(serverEntity.id)!;
+    if (nodeMap.has(serverEntity.entityId)) {
```

Comment:
@aaron-steinfeld Please check this change. @itssharmasandeep Please explain the change.

Reply:
Updated @anandtiwary @aaron-steinfeld

Reply:
So by not using the special value 'id' we just tell apollo not to cache by ref, which is fine. But given this keeps coming up, we really should make the correct change and fix the apollo config to disable this feature - we don't want to ever use it in our current design.

Reply:
If I understood the initial issue correctly though, are you modifying the object returned from the query directly? (If so, removing the caching by id won't resolve that)

Reply:
I tried this https://www.apollographql.com/docs/react/caching/cache-configuration#disabling-normalization earlier, do not remember if I was making a mistake somewhere but it did not work at that time

Reply:
That's the right place to look. Feel free to go ahead with this one in the meantime, but if you have a chance and can put together a PR with the client approach, we can try to dig into what was going wrong.

Reply:
Sure, I will try to do that. Thanks :)